code/modules/surgery/blood_filter.dm

```diff
@@ -57,20 +57,33 @@
 				break
 
 /datum/surgery_step/filter_blood/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
-	if(target.reagents.total_volume || (tox_heal_factor > 0 && target.getToxLoss() > 0))
-		for(var/blood_chem in target.reagents.reagent_list)
-			var/datum/reagent/chem = blood_chem
-			target.reagents.remove_reagent(chem.type, min(chem.volume * chem_purge_factor, 10)) //Removes more reagent for higher amounts
+	var/tox_loss = target.getToxLoss()
+	if(target.reagents.total_volume || (tox_heal_factor > 0 && tox_loss > 0))
+		if(target.reagents.total_volume <= 2)
```

Comment:
I don't like this. This will instantly purge multiple chemicals when its total units are under `2u`, even if the metabolization rate is very low

Reply:
The last leg of blood purging takes an eon to complete even though there's barely any reagents left in the victim's blood, expanding it to this is perfectly in line with the intentions of the PR. *2u of total reagents is not a lot, nothing about this is excessively strong regardless of metabolization speed. 

Take a look at how liver code works - it completely bypasses metabolization speed for the last few units of every toxin and quickly purges it.